diff --git a/src/main.cpp b/src/main.cpp
index f02e072a..36a4e511 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1039,6 +1039,10 @@ setDefaultMode:
     
     m_togBtnAnalog->Disable();
     m_btnTogPTT->Disable();
+    m_btnTogPTT->SetBackgroundColour(wxNullColour);
+#if !defined(__APPLE__)
+    m_btnTogPTT->SetForegroundColour(wxNullColour);
+#endif // !defined(__APPLE__)
     m_togBtnVoiceKeyer->Disable();
 
     // squelch settings
diff --git a/src/ongui.cpp b/src/ongui.cpp
index 44643782..0e8f0e88 100644
--- a/src/ongui.cpp
+++ b/src/ongui.cpp
@@ -1247,8 +1247,8 @@ void MainFrame::OnTogBtnPTTMouseLeave(wxMouseEvent& event)
 {
     m_suppressNextPTTClick_ = false;
 
-    if (!m_btnTogPTT->GetValue() && !g_tx.load(std::memory_order_acquire)
-        && !txChangeoverOccurring_)
+    if (m_btnTogPTT->IsEnabled() && !m_btnTogPTT->GetValue()
+        && !g_tx.load(std::memory_order_acquire) && !txChangeoverOccurring_)
     {
         m_btnTogPTT->SetBackgroundColour(wxSystemSettings::GetColour(wxSYS_COLOUR_BTNFACE));
 #if !defined(__APPLE__)
