diff --git a/cmake/BuildFreeDVBackend.cmake b/cmake/BuildFreeDVBackend.cmake
index da1bc854..86594195 100644
--- a/cmake/BuildFreeDVBackend.cmake
+++ b/cmake/BuildFreeDVBackend.cmake
@@ -2,8 +2,8 @@ include(FetchContent)
 
 FetchContent_Declare(
     freedv_backend
-    GIT_REPOSITORY https://github.com/tmiw/freedv-backend
-    GIT_TAG main
+    GIT_REPOSITORY https://github.com/barjac/freedv-backend
+    GIT_TAG bcj-backend-audio-dev
 )
 
 FetchContent_MakeAvailable(freedv_backend)
