diff --git a/cmake/BuildFreeDVBackend.cmake b/cmake/BuildFreeDVBackend.cmake
index 35210a6..c31c30d 100644
--- a/cmake/BuildFreeDVBackend.cmake
+++ b/cmake/BuildFreeDVBackend.cmake
@@ -2,8 +2,8 @@
 
 FetchContent_Declare(
     freedv_backend
-    GIT_REPOSITORY https://github.com/tmiw/freedv-backend
-    GIT_TAG ms-agc-tweaks
+    GIT_REPOSITORY https://github.com/barjac/freedv-backend
+    GIT_TAG bcj-ms-agc-tweaks-diag
 )
 
 FetchContent_MakeAvailable(freedv_backend)
