diff --git a/cmake/BuildHamlib.cmake b/cmake/BuildHamlib.cmake
index 7dfa6b1f..192fc886 100644
--- a/cmake/BuildHamlib.cmake
+++ b/cmake/BuildHamlib.cmake
@@ -46,3 +46,6 @@ list(APPEND FREEDV_STATIC_DEPS hamlib)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector")
 set(HAMLIB_ADD_DEPENDENCY TRUE)
+
+add_definitions(-DHAMLIB_USE_FRIENDLY_ERRORS)
+add_definitions(-DHAMLIB_CONST_WORKAROUND)
