diff --git a/src/defines.h b/src/defines.h
index e64c60e3..6de51291 100644
--- a/src/defines.h
+++ b/src/defines.h
@@ -89,7 +89,18 @@
 // Level Gauge
 #define FROM_RADIO_MAX       0.8
 #define FROM_MIC_MAX         0.8
-#define LEVEL_BETA           0.99
+
+// Decay rate for the Level meter.
+// Calculated based on desired 12dB/sec decay:
+// -40 = 20 log(x/INT16_MAX)
+// -2 = log(x/INT16_MAX)
+// 10^-2 = x/INT16_MAX
+// 0.01 * INT16_MAX = 327 = x
+//
+// 40/12 = 3.33s to fully decay = 33 timer fires @ DT sec
+// 327 = 32767 * y^33
+// 33rd root of 0.01 = 0.86
+#define LEVEL_BETA           0.86
 
 // TX Attenuation (0.1 dB increments)
 #define TX_ATTENUATION_MIN (-300) /* -30 dB */
diff --git a/src/gui/dialogs/dlg_filter.cpp b/src/gui/dialogs/dlg_filter.cpp
index 2abfda27..371ca471 100644
--- a/src/gui/dialogs/dlg_filter.cpp
+++ b/src/gui/dialogs/dlg_filter.cpp
@@ -639,7 +639,19 @@ void FilterDlg::OnNoiseReductionEnable(wxScrollEvent&) {
 void FilterDlg::OnAgcEnable(wxScrollEvent&) {
     wxGetApp().appConfiguration.filterConfiguration.agcEnabled = m_ckboxAgcEnabled->GetValue();
     g_agcEnabled.store(wxGetApp().appConfiguration.filterConfiguration.agcEnabled, std::memory_order_release); // forces immediate change at pipeline level
+    if (wxGetApp().appConfiguration.filterConfiguration.agcEnabled)
+    {
+        // Mic level should be reset to 0 if AGC is enabled.
+        wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB = 0;
+        m_MicInVol.gaindB = wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB;
+        setGain(&m_MicInVol);
+
+        plotMicInFilterSpectrum();
+        adjRunTimeMicInFilter();
+    }
+
     ExchangeData(EXCHANGE_DATA_OUT);
+    updateControlState();
 }
 
 void FilterDlg::OnBwExpandEnable(wxScrollEvent&) {
@@ -662,7 +674,7 @@ void FilterDlg::updateControlState()
     m_MicInTreble.sliderFreq->Enable(wxGetApp().appConfiguration.filterConfiguration.micInChannel.eqEnable);
     m_MicInTreble.sliderGain->Enable(wxGetApp().appConfiguration.filterConfiguration.micInChannel.eqEnable);
     
-    m_MicInVol.sliderGain->Enable(true);
+    m_MicInVol.sliderGain->Enable(!wxGetApp().appConfiguration.filterConfiguration.agcEnabled);
     
     m_MicInDefault->Enable(wxGetApp().appConfiguration.filterConfiguration.micInChannel.eqEnable);
     
diff --git a/src/main.cpp b/src/main.cpp
index f0cf48ea..44c095cd 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -135,7 +135,8 @@ time_t              g_sync_time;
 constexpr int PLOT_BUF_MULTIPLIER=8;
 GenericFIFO<short>  g_plotDemodInFifo(PLOT_BUF_MULTIPLIER*WAVEFORM_PLOT_BUF);
 GenericFIFO<short>  g_plotSpeechOutFifo(PLOT_BUF_MULTIPLIER*WAVEFORM_PLOT_BUF);
-GenericFIFO<short>  g_plotSpeechInFifo(PLOT_BUF_MULTIPLIER*WAVEFORM_PLOT_BUF);
+GenericFIFO<short>  g_plotSpeechInFifoBeforeEQ(PLOT_BUF_MULTIPLIER*WAVEFORM_PLOT_BUF);
+GenericFIFO<short>  g_plotSpeechInFifoAfterAGC(PLOT_BUF_MULTIPLIER*WAVEFORM_PLOT_BUF);
 
 // Soundcard config
 int                 g_nSoundCards;
@@ -899,7 +900,12 @@ void MainFrame::loadConfiguration_()
     
     // Load AGC state
     g_agcEnabled.store(wxGetApp().appConfiguration.filterConfiguration.agcEnabled, std::memory_order_release);
-    
+    if (wxGetApp().appConfiguration.filterConfiguration.agcEnabled)
+    {
+        // Mic level should be reset to 0 if AGC is enabled.
+        wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB = 0;
+    }
+
     // Load BW expander state
     g_bwExpandEnabled.store(wxGetApp().appConfiguration.filterConfiguration.bwExpandEnabled, std::memory_order_release);
     
@@ -1991,7 +1997,8 @@ int MainFrame::getIdealStationsHeardColumnLength_(int col)
 //----------------------------------------------------------------
 void MainFrame::OnTimer(wxTimerEvent &evt)
 {
-    short speechInPlotSamples[WAVEFORM_PLOT_BUF];
+    short speechInPlotSamplesBeforeEQ[WAVEFORM_PLOT_BUF];
+    short speechInPlotSamplesAfterAGC[WAVEFORM_PLOT_BUF];
     short speechOutPlotSamples[WAVEFORM_PLOT_BUF];
     short demodInPlotSamples[WAVEFORM_PLOT_BUF];
     bool txState = false;
@@ -2056,11 +2063,16 @@ void MainFrame::OnTimer(wxTimerEvent &evt)
       }
       else if (timerId == ID_TIMER_SPEECH_IN)
       {
-          if (g_plotSpeechInFifo.read(speechInPlotSamples, WAVEFORM_PLOT_BUF)) {
-              memset(speechInPlotSamples, 0, WAVEFORM_PLOT_BUF*sizeof(short));
+          if (g_plotSpeechInFifoAfterAGC.read(speechInPlotSamplesAfterAGC, WAVEFORM_PLOT_BUF)) {
+              memset(speechInPlotSamplesAfterAGC, 0, WAVEFORM_PLOT_BUF*sizeof(short));
           }
-          m_panelSpeechIn->add_new_short_samples(speechInPlotSamples, WAVEFORM_PLOT_BUF, 32767);
+          m_panelSpeechIn->add_new_short_samples(speechInPlotSamplesAfterAGC, WAVEFORM_PLOT_BUF, 32767);
           m_panelSpeechIn->refreshData();
+          
+          if (g_plotSpeechInFifoBeforeEQ.read(speechInPlotSamplesBeforeEQ, WAVEFORM_PLOT_BUF))
+          {
+              memset(speechInPlotSamplesBeforeEQ, 0, WAVEFORM_PLOT_BUF*sizeof(short));
+          }
       }
       else if (timerId == ID_TIMER_SPEECH_OUT)
       {
@@ -2116,14 +2128,7 @@ void MainFrame::OnTimer(wxTimerEvent &evt)
              if (m_filterDialog->haveVolumesBeenChanged())
              {
                 auto sliderVal = 0.0;
-                if (txState)
-                {
-                    sliderVal = wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB;
-                }
-                else
-                {
-                    sliderVal = wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB;
-                }
+                sliderVal = wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB;
 
                 if ((sliderVal * 10) != m_sliderMicSpkrLevel->GetValue())
                 {
@@ -2469,8 +2474,8 @@ void MainFrame::OnTimer(wxTimerEvent &evt)
             // peak from this DT sampling period
             int maxSpeechIn = 0;
             for(int i=0; i<WAVEFORM_PLOT_BUF; i++)
-                if (maxSpeechIn < abs(speechInPlotSamples[i]))
-                    maxSpeechIn = abs(speechInPlotSamples[i]);
+                if (maxSpeechIn < abs(speechInPlotSamplesBeforeEQ[i]))
+                    maxSpeechIn = abs(speechInPlotSamplesBeforeEQ[i]);
 
             // peak from last second
             if (maxSpeechIn > m_maxLevel)
@@ -2482,8 +2487,8 @@ void MainFrame::OnTimer(wxTimerEvent &evt)
         if (updated)
         {
             // Peak Reading meter: updates peaks immediately, then slowly decays
-            int maxScaled = (int)(100.0 * ((float)m_maxLevel/32767.0));
-            m_gaugeLevel->SetValue(maxScaled);
+            int maxScaled = m_maxLevel == 0 ? -40 : 20 * std::log10((float)m_maxLevel/32767.0); // log(0) is undefined
+            m_gaugeLevel->SetValue(std::max(-40, maxScaled) + 40); // 1/32767 -> -90dB
             m_maxLevel *= LEVEL_BETA;
         }
     }
@@ -2599,7 +2604,8 @@ void MainFrame::performFreeDVOn_()
         // Reset plot FIFOs
         g_plotDemodInFifo.reset();
         g_plotSpeechOutFifo.reset();
-        g_plotSpeechInFifo.reset();
+        g_plotSpeechInFifoBeforeEQ.reset();
+        g_plotSpeechInFifoAfterAGC.reset();
 
         m_txtCtrlCallSign->SetValue(wxT(""));
         m_lastReportedCallsignListView->DeleteAllItems();
diff --git a/src/ongui.cpp b/src/ongui.cpp
index 140415ba..0bc4b602 100644
--- a/src/ongui.cpp
+++ b/src/ongui.cpp
@@ -898,17 +898,8 @@ void MainFrame::OnTuneAttenContextMenu( wxContextMenuEvent& )
 void MainFrame::OnChangeMicSpkrLevel( wxScrollEvent& )
 {
     auto sliderLevel = (double)m_sliderMicSpkrLevel->GetValue() / 10.0;
-    
-    if (g_tx.load(std::memory_order_acquire))
-    {
-        wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB = sliderLevel;
-        m_newMicInFilter = true;
-    }
-    else
-    {
-        wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB = sliderLevel;
-        m_newSpkOutFilter = true;
-    }
+    wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB = sliderLevel;
+    m_newSpkOutFilter = true;
     
     wxString fmtString = wxString::Format(MIC_SPKR_LEVEL_FORMAT_STR, wxNumberFormatter::ToString((double)sliderLevel, 1), DECIBEL_STR);
     m_txtMicSpkrLevelNum->SetLabel(fmtString);
@@ -1586,26 +1577,8 @@ void MainFrame::togglePTT(void) {
     m_cboReportFrequency->Enable(!newTx);
     m_btnTogTune->Enable(!newTx);
 
-    if (newTx)
-    {
-        micSpeakerBox->SetLabel("Mic &Level");
-
-        m_sliderMicSpkrLevel->SetValue(wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB * 10);
-        wxString fmtString = wxString::Format(MIC_SPKR_LEVEL_FORMAT_STR, wxNumberFormatter::ToString((double)wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB, 1), DECIBEL_STR);
-        m_txtMicSpkrLevelNum->SetLabel(fmtString);
-    }
-    else
-    {
-        micSpeakerBox->SetLabel("Speaker &Level");
-
-        m_sliderMicSpkrLevel->SetValue(wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB * 10);
-        wxString fmtString = wxString::Format(MIC_SPKR_LEVEL_FORMAT_STR, wxNumberFormatter::ToString((double)wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB, 1), DECIBEL_STR);
-        m_txtMicSpkrLevelNum->SetLabel(fmtString);
-    }
-
     CallAfter([&]() {
         txChangeoverOccurring_ = false;
-        m_sliderMicSpkrLevel->Refresh(); // Redraw doesn't happen immediately otherwise in some environments
     });
 
     if (newTx && m_momentaryKeyReleasedDuringChangeover_)
@@ -2102,16 +2075,8 @@ void MainFrame::updateReportingFreqList_()
 void MainFrame::OnResetMicSpkrLevel(wxMouseEvent&)
 {
     auto sliderLevel = 0;
-    if (g_tx.load(std::memory_order_acquire))
-    {
-        wxGetApp().appConfiguration.filterConfiguration.micInChannel.volInDB = sliderLevel;
-        m_newMicInFilter = true;
-    }
-    else
-    {
-        wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB = sliderLevel;
-        m_newSpkOutFilter = true;
-    }
+    wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB = sliderLevel;
+    m_newSpkOutFilter = true;
     
     wxString fmtString = wxString::Format(MIC_SPKR_LEVEL_FORMAT_STR, wxNumberFormatter::ToString((double)sliderLevel, 1), DECIBEL_STR);
     m_txtMicSpkrLevelNum->SetLabel(fmtString);
diff --git a/src/pipeline/TxRxThread.cpp b/src/pipeline/TxRxThread.cpp
index dc03077f..ca8f421a 100644
--- a/src/pipeline/TxRxThread.cpp
+++ b/src/pipeline/TxRxThread.cpp
@@ -90,7 +90,8 @@ extern std::atomic<bool> g_playFileToMicIn;
 extern int g_sfTxFs;
 extern bool g_loopPlayFileToMicIn;
 extern std::atomic<float> g_TxFreqOffsetHz;
-extern GenericFIFO<short> g_plotSpeechInFifo;
+extern GenericFIFO<short> g_plotSpeechInFifoBeforeEQ;
+extern GenericFIFO<short> g_plotSpeechInFifoAfterAGC;
 extern GenericFIFO<short> g_plotDemodInFifo;
 extern GenericFIFO<short> g_plotSpeechOutFifo;
 extern int g_txLevel;
@@ -210,6 +211,29 @@ void TxRxThread::initializePipeline_()
             eitherOrBypassRNNoise);
         pipeline_->appendPipelineStep(eitherOrRNNoiseStep);
 
+        // Resample for plot step (before equalization)
+        auto resampleForPlotStepBeforeEQ = new ResampleForPlotStep(&g_plotSpeechInFifoBeforeEQ);
+        auto resampleForPlotPipelineBeforeEQ = new AudioPipeline(inputSampleRate_, resampleForPlotStepBeforeEQ->getOutputSampleRate());
+#if defined(ENABLE_FASTER_PLOTS)
+        auto resampleForPlotResamplerBeforeEQ = new ResampleStep(inputSampleRate_, resampleForPlotStepBeforeEQ->getInputSampleRate(), true); // need to create manually to get access to "plot only" optimizations
+        resampleForPlotPipelineBeforeEQ->appendPipelineStep(resampleForPlotResamplerBeforeEQ);
+#endif // defined(ENABLE_FASTER_PLOTS)
+        resampleForPlotPipelineBeforeEQ->appendPipelineStep(resampleForPlotStepBeforeEQ);
+
+        auto resampleForPlotTapBeforeEQ = new TapStep(inputSampleRate_, resampleForPlotPipelineBeforeEQ);
+        pipeline_->appendPipelineStep(resampleForPlotTapBeforeEQ);
+
+        // Equalizer step (optional based on filter state)
+        auto equalizerStep = new EqualizerStep(
+            inputSampleRate_, 
+            &g_rxUserdata->micInEQEnable,
+            &g_rxUserdata->sbqMicInBass,
+            &g_rxUserdata->sbqMicInMid,
+            &g_rxUserdata->sbqMicInTreble,
+            &g_rxUserdata->sbqMicInVol,
+            g_rxUserdata->micEqLock);
+        pipeline_->appendPipelineStep(equalizerStep);
+
         // AGC step (optional)
         auto eitherOrProcessAgc = new AudioPipeline(inputSampleRate_, inputSampleRate_);
         auto eitherOrBypassAgc = new AudioPipeline(inputSampleRate_, inputSampleRate_);
@@ -223,17 +247,18 @@ void TxRxThread::initializePipeline_()
             eitherOrBypassAgc);
         pipeline_->appendPipelineStep(eitherOrAgcStep); 
 
-        // Equalizer step (optional based on filter state)
-        auto equalizerStep = new EqualizerStep(
-            inputSampleRate_, 
-            &g_rxUserdata->micInEQEnable,
-            &g_rxUserdata->sbqMicInBass,
-            &g_rxUserdata->sbqMicInMid,
-            &g_rxUserdata->sbqMicInTreble,
-            &g_rxUserdata->sbqMicInVol,
-            g_rxUserdata->micEqLock);
-        pipeline_->appendPipelineStep(equalizerStep);
-        
+        // Resample for plot step (after AGC)
+        auto resampleForPlotStepAfterAGC = new ResampleForPlotStep(&g_plotSpeechInFifoAfterAGC);
+        auto resampleForPlotPipelineAfterAGC = new AudioPipeline(inputSampleRate_, resampleForPlotStepAfterAGC->getOutputSampleRate());
+#if defined(ENABLE_FASTER_PLOTS)
+        auto resampleForPlotResamplerAfterAGC = new ResampleStep(inputSampleRate_, resampleForPlotStepAfterAGC->getInputSampleRate(), true); // need to create manually to get access to "plot only" optimizations
+        resampleForPlotPipelineAfterAGC->appendPipelineStep(resampleForPlotResamplerAfterAGC);
+#endif // defined(ENABLE_FASTER_PLOTS)
+        resampleForPlotPipelineAfterAGC->appendPipelineStep(resampleForPlotStepAfterAGC);
+
+        auto resampleForPlotTapAfterAGC = new TapStep(inputSampleRate_, resampleForPlotPipelineAfterAGC);
+        pipeline_->appendPipelineStep(resampleForPlotTapAfterAGC);
+
         // Take TX audio post-equalizer and send it to RX for possible monitoring use.
         if (equalizedMicAudioLink_ != nullptr)
         {
@@ -243,18 +268,6 @@ void TxRxThread::initializePipeline_()
             auto micAudioTap = new TapStep(inputSampleRate_, micAudioPipeline);
             pipeline_->appendPipelineStep(micAudioTap);
         }
-                
-        // Resample for plot step
-        auto resampleForPlotStep = new ResampleForPlotStep(&g_plotSpeechInFifo);
-        auto resampleForPlotPipeline = new AudioPipeline(inputSampleRate_, resampleForPlotStep->getOutputSampleRate());
-#if defined(ENABLE_FASTER_PLOTS)
-        auto resampleForPlotResampler = new ResampleStep(inputSampleRate_, resampleForPlotStep->getInputSampleRate(), true); // need to create manually to get access to "plot only" optimizations
-        resampleForPlotPipeline->appendPipelineStep(resampleForPlotResampler);
-#endif // defined(ENABLE_FASTER_PLOTS)
-        resampleForPlotPipeline->appendPipelineStep(resampleForPlotStep);
-
-        auto resampleForPlotTap = new TapStep(inputSampleRate_, resampleForPlotPipeline);
-        pipeline_->appendPipelineStep(resampleForPlotTap);
       
         // FreeDV TX step (analog leg)
         auto doubleLevelStep = new LevelAdjustStep(inputSampleRate_, +[]() FREEDV_NONBLOCKING { return (float)2.0; });
diff --git a/src/topFrame.cpp b/src/topFrame.cpp
index 1373b95b..594bff93 100644
--- a/src/topFrame.cpp
+++ b/src/topFrame.cpp
@@ -879,8 +879,8 @@ TopFrame::TopFrame(wxWindow* parent, wxWindowID id, const wxString& title, const
     //------------------------------
     levelBox = new TintedGroupBox(m_panel, _("Level"), wxHORIZONTAL, 1);
 
-    m_gaugeLevel = new wxGauge(levelBox, wxID_ANY, 100, wxDefaultPosition, wxSize(135,15), wxGA_SMOOTH);
-    m_gaugeLevel->SetToolTip(_("Peak of From Radio in Rx, or peak of From Mic in Tx mode."));
+    m_gaugeLevel = new wxGauge(levelBox, wxID_ANY, 40, wxDefaultPosition, wxSize(100,15), wxGA_SMOOTH); // log scale, -40 dB to 0 dB
+    m_gaugeLevel->SetToolTip(_("RX: Peak level of radio's audio output, TX: Peak level of microphone audio as recorded by FreeDV (before AGC/level settings)."));
 #if defined(__WXGTK__) && defined(HAS_GTK3)
     gtk_style_context_add_class(gtk_widget_get_style_context(GTK_WIDGET(m_gaugeLevel->GetHandle())), "freedv-meter");
 #endif // defined(__WXGTK__) && defined(HAS_GTK3)
