diff --git a/src/ongui.cpp b/src/ongui.cpp
index 57a15950..eaaa0760 100644
--- a/src/ongui.cpp
+++ b/src/ongui.cpp
@@ -1113,7 +1113,7 @@ void MainFrame::OnTogBtnPTTRightClick( wxContextMenuEvent& )
 //-------------------------------------------------------------------------
 // OnTogBtnPTT ()
 //-------------------------------------------------------------------------
-void MainFrame::OnTogBtnPTT (wxCommandEvent& event)
+void MainFrame::OnTogBtnPTT (wxCommandEvent&)
 {
     if (vk_state == VK_TX)
     {
@@ -1121,10 +1121,9 @@ void MainFrame::OnTogBtnPTT (wxCommandEvent& event)
         VoiceKeyerProcessEvent(VK_SPACE_BAR);
     }
     else
-    {        
+    {
         togglePTT();
     }
-    event.Skip();
 }
 
 void MainFrame::togglePTT(void) {
@@ -1333,7 +1332,7 @@ void MainFrame::togglePTT(void) {
     }
 
     auto newTx = m_btnTogPTT->GetValue();
-    if (wxGetApp().rigPttController != nullptr && wxGetApp().rigPttController->isConnected()) 
+    if (wxGetApp().rigPttController != nullptr && wxGetApp().rigPttController->isConnected())
     {
         wxGetApp().rigPttController->ptt(newTx);
     }
