diff --git a/src/ongui.cpp b/src/ongui.cpp
index 9a028be1..ae4355b2 100644
--- a/src/ongui.cpp
+++ b/src/ongui.cpp
@@ -1703,7 +1703,8 @@ void MainFrame::OnResetMicSpkrLevel(wxMouseEvent&)
         wxGetApp().appConfiguration.filterConfiguration.spkOutChannel.volInDB = sliderLevel;
         m_newSpkOutFilter = true;
     }
-    
+
+    m_sliderMicSpkrLevel->SetValue(sliderLevel);
     wxString fmtString = wxString::Format(MIC_SPKR_LEVEL_FORMAT_STR, wxNumberFormatter::ToString((double)sliderLevel, 1), DECIBEL_STR);
     m_txtMicSpkrLevelNum->SetLabel(fmtString);
 }
